packages/block-editor/src/components/block-list/use-block-props/index.js

```diff
@@ -54,10 +57,14 @@ const BLOCK_ANIMATION_THRESHOLD = 200;
  *                                           the ref if one is defined.
  * @param {Object}  options                  Options for internal use only.
  * @param {boolean} options.__unstableIsHtml
+ * @param {boolean} options.isDisabled       Whether the block should be disabled.
```

Comment:
Another approach here would be a `options.hasOverlay` flag and bundle the call to useBlockOverlay here. I might switch to that as it avoids a lot of duplication.